site/tests/app/controllers/student/SubmissionControllerTester.php

```diff
@@ -783,48 +783,6 @@ public function testEmptyPost() {
         $this->assertFalse($return['status'] == 'success');
     }
 
-    public function testErrorNotSetCsrfToken() {
-        $_POST['csrf_token'] = null;
-        $return = $this->runController();
-        $this->assertTrue($return['status'] == 'fail');
-        $this->assertEquals("Invalid CSRF token.", $return['message']);
-        $this->assertFalse($return['status'] == 'success');
-    }
-
-    /**
-     * Test that one must have at least full grading access to change the active submission version a user
-     */
-    public function testUpdateSubmissionVersionPermission() {
-        $_REQUEST['action'] = 'update';
-        $_REQUEST['ta'] = 'true';
-        $return = $this->runController($this->createMockCore($user_config = array('access_full_grading' => false)));
-        $this->assertTrue($return['status'] == 'fail');
-        $this->assertEquals("You do not have access to that page.", $return['message']);
-        unset($_REQUEST['ta']);
-    }
-
-    /**
-     * Test that one must be at least a full access grader to delete split items
-     */
-    public function testDeleteSplitItemPermission() {
-        $_REQUEST['action'] = 'delete_split';
-        $return = $this->runController($this->createMockCore(array('csrf_token' => true), array('access_full_grading' => false)));
-        $this->assertTrue($return['status'] == 'fail');
-        $this->assertFalse($return['status'] == 'success');
-        $this->assertEquals("You do not have access to that page.", $return['message']);
-    }
```

Comment:
Why are we deleting these two functions permission check functions?

Reply:
Because the router checks access before calling the function. The function itself doesn't check that. If we really want to have that check we may need to do that in a more e2e fashion (Create a request and feed the router and see if it is denied).

Reply:
Ah, now I get it. I misunderstood what was going on here and was thinking (wrongly) that we could check these permissions without using the router. I'm fine with these getting removed.

I do agree that we probably want to add some concept of integration tests (not full e2e) for these controllers that center around permissions which should be easier to do across the code-base with the Access + Router checking it instead of it being baked into the method.